pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -273,22 +281,48 @@ func updateFuncHandler(oldObj interface{}, newObj interface{}, object client.Obj
 	return nil
 }
 
-func deleteFuncHandler(obj interface{}, object client.ObjectKey, c chan<- INotify) error {
+// toFFCfg attempts to covert unstructured payload to configurations
+func toFFCfg(object interface{}) (*v1alpha1.FeatureFlagConfiguration, error) {
```

Comment:
if I add the following test to the `Test_toFFCfg` function:
```
{
			name: "Wrong resource type",
			input: toUnstructured(t, corev1.Pod{
				TypeMeta: v1.TypeMeta{
					Kind: "Pod",
				},
			}),
			wantErr: true,
		},
```
the test fails as there is no error returned, this is expected behaviour as the marshalling would not fail, however, IMO it should. We should add an if statement to ensure that the apiversion and kind match at this point, of course this would require the k8s api to return us the wrong object, however I think its worthwhile covering this case as it currently passes with any object

Reply:
Yes, noticed this while testing the function. However, I will keep this as it is. The reason is that if we try to validate, we have to enforce validation criteria. API version is only one such validation. So it is best to allow the caller to perform such validation based on its requirement and have this common logic of parsing here.